core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import alluxio.collections.Pair;
+import alluxio.worker.block.meta.StorageDir;
+import alluxio.worker.block.meta.StorageTier;
+
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class holds the metadata information of a block store.
+ * <p>
+ * TODO(bin): Use proto buf to represent this information.
+ */
+@ThreadSafe
+public final class DefaultBlockStoreMeta implements BlockStoreMeta {
+  // TODO(bin): The following two fields don't need to be computed on the creation of each
+  // {@link BlockStoreMeta} instance.
+
+  /** Mapping from storage tier alias to capacity bytes. */
+  private final Map<String, Long> mCapacityBytesOnTiers = new HashMap<>();
+
+  /** Mapping from storage tier alias to used bytes. */
+  private final Map<String, Long> mUsedBytesOnTiers = new HashMap<>();
+
+  /** Mapping from storage tier alias to capacity bytes. */
+  private final Map<String, List<Long>> mBlockIdsOnTiers;
+
+  /** Mapping from storage dir tier and path to total capacity. */
+  private final Map<Pair<String, String>, Long> mCapacityBytesOnDirs = new HashMap<>();
+
+  /** Mapping from storage dir tier and path to used bytes. */
+  private final Map<Pair<String, String>, Long> mUsedBytesOnDirs = new HashMap<>();
+
+  /**
+   * Note: This is only available in {@link DefaultBlockStoreMeta#getBlockStoreMetaFull}.
```

Comment:
NIT: for `@Override` methods, you don't need to provide javadoc unless it is different from the method we override